torch/csrc/autograd/input_metadata.h

```diff
@@ -6,6 +6,14 @@
 
 namespace torch { namespace autograd {
 
+inline int get_tensor_device(const at::Tensor& tensor) {
+  if (tensor.is_cuda() ||
```

Comment:
would it  be difficult to just rewrite the places where you need this to do this all in terms of Devices (`tensor.device()`) ?

Reply:
This knowledge of "has device", and its way to check, is typical of autograd, so I think it makes sense to have a gating API on the autograd side.